internal/cmd/io_binary_test.go

```diff
@@ -0,0 +1,72 @@
+package cmd
+
+import (
+	"context"
+	"net/url"
+	"reflect"
+	"testing"
+)
+
+func Test_newBinaryCmd_Key_Value_Pair(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	type config struct {
+		name      string
+		urlString string
+		expected  string
+	}
+
+	tests := []*config{
```

Comment:
Can we receive either form?

Reply:
Or rather better question, do we expect to as I assume the client could pass in whatever

Reply:
per offline discussion, it'd make sense to join `uri.Host` and `uri.Path`, so that `sheme://path/to/binary` and `scheme:///path/to/binary` resolve to the same path